### PR TITLE
Chore/mock out filesystem

### PR DESCRIPTION
**Pull request by sighphyre**
Patches the mock file system to behave more realistically, use that in some tests

There's also a behaviour change when loading bootstrap. The test says that should return an empty string when the file isn't found but there's no actual check for a missing file so presumably this crashes with an exception in live. Not actually sure how this is passing, but presumably another test is putting an empty file on the file system

### CODE REVIEW

File: src/Unleash/Utilities/ToggleBootstrapFileProvider.cs [hunk 1]
```diff
@@ -15,7 +16,14 @@ internal ToggleBootstrapFileProvider(string filePath, UnleashSettings settings)
...
         {
-            return settings.FileSystem.ReadAllText(filePath);
+            try
+            {
+                return settings.FileSystem.ReadAllText(filePath);
+            }
+            catch (FileNotFoundException)
+            {
+                return string.Empty;
+            }
```
Comment (sighphyre) [inline src/Unleash/Utilities/ToggleBootstrapFileProvider.cs:26]: [Docs](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.readalltext?view=net-9.0) say this raises a FileNotFoundException if this isn't found, which matches my experience of this API. I've chosen to make it work like the test says it should but I honestly have no idea how this ever passed. Pretty sure it shouldn't have

Comment (daveleek) [inline src/Unleash/Utilities/ToggleBootstrapFileProvider.cs:26]: Maybe this used to use the old mock thing under the hood that does things weirdly, and the test verified the incorrect mock 🤷

File: tests/Unleash.Tests/Mock/MockFileSystem.cs [hunk 2]
```diff
@@ -1,32 +1,70 @@
...
 namespace Unleash.Tests.Mock
 {
     class MockFileSystem : IFileSystem
     {
+        internal readonly Dictionary<string, string> _fileSystem = new();
+        public Encoding Encoding => Encoding.UTF8;
+
         public bool FileExists(string path)
         {
-            return true;
```
Comment (daveleek) [inline tests/Unleash.Tests/Mock/MockFileSystem.cs:10]: Be gone, and never come back

File: src/Unleash/Utilities/ToggleBootstrapFileProvider.cs (hunk 1 above)
Comment (sighphyre) [reply src/Unleash/Utilities/ToggleBootstrapFileProvider.cs:26]: I thought about that but this seemed to call out to the real fs. If there's a swap out for the mock fs I can't spot it
